test/integration/kubernetes_deploy_test.rb

```diff
@@ -376,20 +383,30 @@ def test_wait_false_still_waits_for_priority_resources
   def test_deployment_with_progress_times_out_for_short_duration
     # The deployment adds a short progressDeadlineSeconds and attepts to deploy a container
     # which sleeps and cannot fulfill the readiness probe causing it to timeout
-    result = deploy_fixtures("long-running", subset: ['undying-deployment.yml.erb']) do |fixtures|
+    result, error = deploy_fixtures_with_error("long-running", subset: ['undying-deployment.yml.erb']) do |fixtures|
       deployment = fixtures['undying-deployment.yml.erb']['Deployment'].first
       deployment['spec']['progressDeadlineSeconds'] = 10
       container = deployment['spec']['template']['spec']['containers'].first
       container['readinessProbe'] = { "exec" => { "command" => ['- ls'] } }
     end
     assert_deploy_failure(result)
+    assert_equal error, :timeout
 
     assert_logs_match_all([
       'Deployment/undying: TIMED OUT (progress deadline: 10s)',
       'Timeout reason: ProgressDeadlineExceeded'
     ])
   end
 
+  def test_failed_deployment_error_nil_when_not_timeout
+    result, error = deploy_fixtures_with_error("invalid", subset: ["cannot_run.yml"]) do |fixtures|
+      container = fixtures["cannot_run.yml"]["Deployment"].first["spec"]["template"]["spec"]["containers"].first
+      container["image"] = "some-invalid-image:badtag"
+    end
+    assert_deploy_failure(result)
+    refute_equal error, :timeout
```

Comment:
The name of the test suggests that the `error` is `nil` when it didn't time out, but then it doesn't test for it.

In case we want to make this also work in the future (when we maybe support other errors than `:timeout`) do we want to make the check `assert_equal error, nil`? or change the name of the test?

Reply:
I think you're right,  `assert_nil error` makes sense here. I also think the test name matches the current behavior so should stay put. If / when we add more conditions we'll need to change what we assert here and the test name.